src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -189,11 +198,17 @@ public bool Equals(EmitOptions other)
                 this.RuntimeMetadataVersion == other.RuntimeMetadataVersion &&
                 this.TolerateErrors == other.TolerateErrors &&
                 this.IncludePrivateMembers == other.IncludePrivateMembers &&
-                this.Instrument == other.Instrument;
+                this.InstrumentationKinds == other.InstrumentationKinds;
```

Comment:
@drognanar changed the comparison to use SequenceEqual, which is more meaningful here. Therer is still a question about whether equality should allow differences in order of instrumentation kinds.